src/libcore/fmt/mod.rs

```diff
@@ -846,7 +846,7 @@ macro_rules! tuple {
             fn fmt(&self, f: &mut Formatter) -> Result {
                 try!(write!(f, "("));
                 let ($(ref $name,)*) = *self;
-                let mut n = 0i;
+                let mut n = 0is;
```

Comment:
First usage that needs annotation. Even then, this is used for 3 states, which would be only 2 if we were to special-case one-element tuples. Feels saner that way IMO.
We can do it without adding another case to the macro, too, *and* while removing the `peel` helper macro.
Something like `$first, $($name),*`.